kong/plugins/galileo/alf.lua

```diff
@@ -80,6 +80,22 @@ local function hash_to_array(t)
   return arr
 end
 
+-- Calculate an approximation of header size (it doesn't calculate white 
+-- space that may be sorrounding values, other than that it's accurate)
+local function calculate_headers_size(request_line, headers)
+  local size = 0
+  for k, v in pairs(headers) do
+    if type(v) == "table" then
+      for _, y in ipairs(v) do
+        size = size + #k + 2 + #tostring(y) + 2 --First 2 is semicolon + space
+      end
+    else
+      size = size + #k + 1 + 1 + #tostring(v) + 2 --First 2 is semicolon + space
+    end
+  end
+  return #request_line + 2 + size + 4 -- 2 it's \r\n, 4 it's trailing \r\n\r\n
```

Comment:
I think you already counting one `\r\n` when you calculating the size of last header, so should be final sum be

`#request_line + 2 + size + 2`